src/imageops/sample.rs

```diff
@@ -328,6 +328,50 @@ pub fn sample_bilinear<P: Pixel>(
     )
 }
 
+/// Sample from an image using coordinates in [0,1], taking the nearest coordinate.
+pub fn sample_nearest<P: Pixel>(
+    img: &impl GenericImageView<Pixel = P>,
+    u: f32,
+    v: f32,
+) -> Option<P> {
+    if ![u, v].iter().all(|c| (0.0..=1.0).contains(c)) {
+        return None;
+    }
+
+    let (w, h) = img.dimensions();
+    if w == 0 || h == 0 {
+        return None;
+    }
```

Comment:
This check is redundant with the one below and can be removed
```suggestion
```

Reply:
It's not completely redundant, in debug mode it will cause a panic due to overflow in the following lines with `w-1`, but if that's changed to a `saturating_sub(1)` then it's fine to remove